### PR TITLE
[PB][Docs] PB-35: PageBuilder integration overview

### PR DESCRIPTION
**Pull request by bdenham**
## Description

Adds documentation for the Page Builder integration project, both overview and parsing / rendering details.

## Related Issue

https://jira.corp.magento.com/browse/PB-35

### Verification Stakeholders
@davemacaulay 

### Verification Steps

1. Cd to pwa-docs
3. Run: `npm run develop`.
4. Navigate to http://localhost:3000/page-builder/overview/ to view preview.

## Checklist
* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: pwa-devdocs/src/_drafts/page-builder/how-to-add-your-content-type.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+We utilise a special [master format parser](master-format-parser.md) within our Page Builder integration to easily decompose data within the master format for use in a PWA Studio React component.
+
+## Prerequisites 
+
+- Have PWA Studio setup and running in your development environment.
+- Have a working Page Builder component, with the master format implemented.
+- Be able to render and visualise your content type on the existing Luma store front.
+
+## Structure
+As we require some additional steps compared to a normal we have some additional files associated with each content type, in example our `Text` content types component structure looks like the following:
```
Comment (danmooney2) [inline pwa-devdocs/src/_drafts/page-builder/how-to-add-your-content-type.md]: a normal _what_? getting déjà vu

File: pwa-devdocs/src/_drafts/page-builder/how-to-add-your-content-type.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  - `__tests__`
+  - `configAggregator.js`
+  - `index.js`
+  - `text.css`
+  - `text.js`
+
+`__tests__` & `index.js` are just part of PWA Studio, and won't be discussed here.
+
+### `configAggreagtor.js`
+The config aggregator provides an interface which accepts the HTML master format for a specific content type, decomposes it and returns an flat object of the properties associated with the content type.
```
Comment (danmooney2) [inline pwa-devdocs/src/_drafts/page-builder/how-to-add-your-content-type.md]: a flat object

File: pwa-devdocs/src/_drafts/page-builder/master-format-parser.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+title: Page Builder master format parser
+---
+
+As part of the Page Builder / PWA integration we implemented a system in which we can convert Page Builder's master format into a structured format which React and PWA Studio could understand. We did this on the client side to ensure compatibility with all various hosting methods of Magento Commerce currently available.
+
+# Workflow
+
+1. Page Builder content is fetched as part of an entity request to GraphQL (CMS Page, Product Description etc).
+2. `<RichContent />` is employed to determine what type of content this is, using a simple pattern recognition to determine if the content is Page Builder. If the content does not include Page Builder content we opt to render this out as is.
+3. Master format is passed to our dedicated `<PageBuilder />` which in term parses the HTML into objects.
```
Comment (danmooney2) [inline pwa-devdocs/src/_drafts/page-builder/master-format-parser.md]: in turn

File: pwa-devdocs/src/_drafts/page-builder/overview/index.md [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
+
+![Page Builder Integration Details](PageBuilderIntegrationDetails.svg)
+
+1. The Venia app uses a GraphQL query to get CMS content from your Magento store (pages and blocks) and passes the HTML to the `RichContent` component to begin processing.
+2. The `RichContent` component determines if the HTML contains Page Builder content, using simple pattern recognition. If the HTML does not include Page Builder content, it is returned to Venia and rendered out as is. If the HTML does include Page Builder content, the HTML (master format) is passed to the `PageBuilder` component.
+3. The `PageBuilder` component passes the master format HTML to the `parseStorageHTML` function.
+4. The `parseStorageHtml` function uses a `TreeWalker` to parse all nodes within the master format and create an object tree of all the content types with their associated data. The object tree is passed to the `ContentTypeFactory`.
```
Comment (davemacaulay) [inline pwa-devdocs/src/_drafts/page-builder/overview/index.md]: The ordering here is a little off, `parseStorageHtml` actually calls the config aggregator and passes the configuration (properties) to the factory to then be passed into the component.

Comment (bdenham) [reply pwa-devdocs/src/_drafts/page-builder/overview/index.md]: Got it.

File: pwa-devdocs/src/_drafts/page-builder/how-to-add-your-content-type.md (hunk 1 above)
Comment (bdenham) [reply pwa-devdocs/src/_drafts/page-builder/how-to-add-your-content-type.md]: Sorry. This was content that shouldn't have been included in the PR.

File: pwa-devdocs/src/_drafts/page-builder/how-to-add-your-content-type.md (hunk 2 above)
Comment (bdenham) [reply pwa-devdocs/src/_drafts/page-builder/how-to-add-your-content-type.md]: Shouldn't have been included in PR.

File: pwa-devdocs/src/_drafts/page-builder/master-format-parser.md (hunk 3 above)
Comment (bdenham) [reply pwa-devdocs/src/_drafts/page-builder/master-format-parser.md]: Disregard.

**Pull request merged**